src/applications/disability-benefits/all-claims/components/OfficialReport.jsx

```diff
@@ -0,0 +1,304 @@
+import { POLICE_REPORT_LOCATION_FIELDS } from '../constants';
+
+export function shouldShowPoliceDataModal(event) {
+  return (
+    event &&
+    !event?.otherReports?.police &&
+    POLICE_REPORT_LOCATION_FIELDS.some(
+      field => typeof event[field] === 'string' && event[field].trim() !== '',
+    )
+  );
+}
+
+const OfficialReport = props => {
+  const formDomRef = useRef(null);
+  const modalRef = useRef(null);
+  const alertRef = useRef(null);
+  const submitButtonRef = useRef(null);
+
+  const searchParams = getArrayUrlSearchParams();
+  const isEdit = !!searchParams.get('edit');
+  const isAdd = !!searchParams.get('add');
+  const index = parseInt(props.pagePerItemIndex, 10);
+
+  const [tempData, setTempData] = useState(props.data || {});
+  const [showModal, setShowModal] = useState(false);
+  const [showAlert, setShowAlert] = useState(false);
+
+  const handlers = {
+    shouldShowPoliceDataModal,
+    onModalOrContinue: ({ formData }) => {
+      if (shouldShowPoliceDataModal(formData)) {
+        setShowModal(true);
+        return;
+      }
+
+      if (!formDomRef.current) return;
+      setTempData(formData);
+      formDomRef.current.submitForm();
+    },
+    onSubmit: ({ formData }) => {
+      if (handlers.shouldShowPoliceDataModal(formData)) {
+        setShowModal(true);
+        setTempData(formData);
+        return;
+      }
+
+      props.onChange(formData);
+      props.onSubmit({ formData });
+    },
+    onCancelModal: () => {
+      setShowModal(false);
+      if (!props.data) return;
+      const restoredData = {
+        ...tempData,
+        otherReports: { ...tempData.otherReports, police: true },
+      };
+
+      setTempData(restoredData);
+    },
+    onCancelAlert: () => {
+      setShowAlert(false);
+    },
+    removePoliceReport: () => {
+      setShowModal(false);
+
+      const updatedData = { ...tempData };
+      POLICE_REPORT_LOCATION_FIELDS.forEach(field => {
+        updatedData[field] = '';
+      });
+      updatedData.otherReports = { ...updatedData.otherReports, police: false };
+
+      setTempData(updatedData);
+      props.onChange(updatedData);
+      setShowAlert(true);
+    },
+    handleChange: newData => {
+      setTempData(newData);
+    },
+  };
+
+  const { schema, uiSchema } = useEditOrAddForm({
+    isEdit,
+    schema: isRelatedToMST(props.fullData)
+      ? officialReportMst.schema
+      : officialReport.schema,
+    uiSchema: isRelatedToMST(props.fullData)
+      ? officialReportMst.uiSchema
+      : officialReport.uiSchema,
+    data: tempData,
+    fullData: props.fullData,
+    onChange: handlers.handleChange,
+    onSubmit: handlers.onModalOrContinue,
+    index: props.pagePerItemIndex ? index : null,
+    arrayPath: props.arrayBuilder.arrayPath,
+  });
+
+  useEffect(
+    () => {
+      if (showModal && modalRef.current) {
+        setTimeout(() => {
+          const modalHeading = modalRef.current.shadowRoot?.querySelector(
+            '#heading',
```

Comment:
Confused why this is running the query selector off the shadow root. Also I think technically you can have elements with the same ID in HTML it's just bad practice, so small chance this could grab the wrong heading?

Not blocking on the selector but I think we should doc this before merging in case someone else ends up working on it

Reply:
Honestly, after a number of unsuccessful, although probably arguably more conventional, ways to direct focus on the modal's heading, once I found one that worked, I moved on.